chainermn/link.py

```diff
@@ -0,0 +1,191 @@
+import chainer
+import chainermn
+import chainermn.communicators
+import chainermn.functions
+import chainermn.functions.point_to_point_communication
+
+
+class MultiNodeChainGroup(chainer.ChainList):
+    """Combining multiple non-connected components of computational graph.
+
+    This class combines each ``chainer.Chain``, which represents one of the
+    non-connected component in compuational graph. In ``__call__()``,
+    the returned object of ``chainer.Chain`` (which represents pointer)
+    are passed to the next ``chainer.Chain``, in order to retain the
+    computational graph connected and make backprop work properly.
+
+    Users add each ``chainer.Chain`` by ``add_link()`` method. Each chain
+    is invoked in forward computation according to the order they are added,
+    and in backward computation according to the reversed order.
+
+    .. admonition:: Example
+
+        This is a simple example of the model which sends its outputs to
+        rank=1 machine::
+
+            import chainer
+            import chainer.functions as F
+            import chainermn
+
+
+            class SimpleModelInst(chainer.Chain):
```

Comment:
I didn't understand what `Inst` does mean. How about `SimpleModelSub` or something like that?